src/omgcli.ts

```diff
@@ -358,6 +358,69 @@ export class OMGCLI {
     return await this.childChain.submitTransaction(signedTxn);
   }
 
+  async sendTypedTx(owner: String, currency: String, amount: number) {
+    const payments = [
+      {
+        owner: owner,
+        currency: currency,
+        amount: amount
+      }
+    ];
+
+    const fee = {
+      currency: transaction.ETH_CURRENCY
+    };
+
+    const createdTx = await this.childChain.createTransaction({
+      owner: owner,
+      payments,
+      fee
+    });
+
+    const txTypedData = this.childChain.signTypedData(
+      createdTx.transactions[0],
+      [this.txOptions.privateKey]
+    );
+    return await this.childChain.submitTyped(txTypedData);
+  }
+
+  /*
+   * utxo_pos needs to be fee enabled and have enough balance to pay the fees
+   */
+  async generateTx(from: String, utxo_pos: number) {
+    const utxo = await this.getUTXO(from, utxo_pos);
+
+    if (utxo) {
+      let inputs: any = [{}];
+      inputs[0].blknum = utxo.blknum;
+      inputs[0].txindex = utxo.txindex;
+      inputs[0].oindex = utxo.oindex;
+
+      let outputs: any = [{}];
+      outputs[0].outputGuard = utxo.owner;
+      outputs[0].currency = utxo.currency;
+      outputs[0].amount = utxo.amount;
+      outputs[0].outputType = 1;
+
+      let tx: any = {
+        transactionType: 1,
+        inputs,
+        outputs,
+        metadata:
+          "0x0000000000000000000000000000000000000000000000000000000000001337"
+      };
+      const newTx = await this.addFeesToTx(tx);
+      if (newTx) {
```

Comment:
Non-blocking `try/catch` could be suited here. 

``` typescript
let newTx
try {
  newTx = await ... 
} catch {
  console.log( .... ) 
}

Reply:
Why is it better to have a none blocking try/catch block? Should this be handled at a higher level?

Reply:
Sorry. I missed the word `but` in there. 😆  (Non-blocking comment, but ... )

I had in mind the scenario of failure to fetch fees. Could be better handled in `addFeesToTx` perhaps. It just seemed to me that the `if` statement could be replaced by a `try/catch` somewhere along the chain - but nothing too essential here!